ardent-andromedas/src/bot/discord_event.py

```diff
@@ -0,0 +1,159 @@
+
+    Attributes
+    ----------
+        id (int): The unique identifier of the text channel.
+        name (str): The name of the text channel.
+
+    """
+
+    id: int
+    name: str
+
+
+@dataclass
+class SerializableMember:
+    """Serializable representation of a Discord member.
+
+    Attributes
+    ----------
+        id (int): The unique identifier of the member.
+        name (str): The username of the member.
+        display_name (str): The display name of the member.
+        roles (list[int]): List of role IDs the member has.
+        guild_id (int): The ID of the guild the member belongs to.
+        avatar (str): The URL of the member's avatar.
+        color (str): The color associated with the member's top role.
+
+    """
+
+    id: int
+    name: str
+    display_name: str
+    roles: list[int]
+    guild_id: int
+    avatar: str
+    color: str
+
+
+@dataclass
+class DiscordEvent:
+    """A class representing a Discord event with relevant information.
+
+    Attributes
+    ----------
+        type (EventType): The type of the Discord event.
+        timestamp (datetime): The timestamp when the event occurred.
+        guild (SerializableGuild): Serializable representation of the Discord guild.
+        channel (SerializableTextChannel): Serializable representation of the text channel.
+        member (SerializableMember | FakeUser): Serializable representation of the member or a FakeUser.
+        content (str): The content or message associated with the event.
+        reaction_image (Optional[bytes]): The image data of the reaction emoji, if applicable.
+
+    """
+
+    type: EventType
+    timestamp: datetime
+    guild: SerializableGuild
+    channel: SerializableTextChannel
+    member: SerializableMember | FakeUser
+    content: str
+    reaction_image: bytes | None = None
+
+    @classmethod
+    def from_discord_objects(
+        cls,
+        type: EventType,
+        timestamp: datetime,
+        guild: Optional["discord.Guild"],
+        channel: Optional["discord.TextChannel"],
+        member: Optional["discord.Member"],
+        content: str,
+        reaction_image: bytes | None = None,
+    ) -> "DiscordEvent":
+        return cls(
+            type=type,
+            timestamp=timestamp,
+            guild=SerializableGuild(guild.id, guild.name, guild.verification_level.value) if guild else None,
+            channel=SerializableTextChannel(channel.id, channel.name)
+            if channel and hasattr(channel, "id") and hasattr(channel, "name")
+            else None,
+            member=SerializableMember(
+                member.id,
+                member.name,
+                member.display_name,
+                [role.id for role in member.roles],
+                member.guild.id,
+                str(member.avatar.url) if member.avatar else None,
+                str(member.color),
+            )
+            if member
+            else None,
```

Comment:
to make this easier to read
```suggestion
            channel=(
                SerializableTextChannel(channel.id, channel.name)
                if channel and hasattr(channel, "id") and hasattr(channel, "name")
                else None
            ),
            member=(
                SerializableMember(
                    member.id,
                    member.name,
                    member.display_name,
                    [role.id for role in member.roles],
                    member.guild.id,
                    str(member.avatar.url) if member.avatar else None,
                    str(member.color),
                )
                if member
                else None
            ),
```